nav2_behavior_tree/test/plugins/decorator/distance_controller_test.cpp

```diff
@@ -0,0 +1,177 @@
+    config_->blackboard->set<bool>("path_updated", false);
+    config_->blackboard->set<bool>("initial_pose_received", false);
+
+    transform_handler_->activate();
+    transform_handler_->waitForTransform();
+  }
+
+  static void TearDownTestCase()
+  {
+    transform_handler_->deactivate();
+    delete transform_handler_;
+    delete config_;
+    delete dummy_node_;
+    transform_handler_ = nullptr;
+    config_ = nullptr;
+    dummy_node_ = nullptr;
+  }
+
+  void SetUp()
+  {
+    node_ = new nav2_behavior_tree::DistanceController("distance_controller", *config_);
+    node_->setChild(dummy_node_);
+  }
+
+  void TearDown()
+  {
+    node_ = nullptr;
+  }
+
+protected:
+  static nav2_behavior_tree::TransformHandler * transform_handler_;
+  static BT::NodeConfiguration * config_;
+  static nav2_behavior_tree::DistanceController * node_;
+  static nav2_behavior_tree::DummyNode * dummy_node_;
+};
+
+nav2_behavior_tree::TransformHandler * DistanceControllerTestFixture::transform_handler_ = nullptr;
+BT::NodeConfiguration * DistanceControllerTestFixture::config_ = nullptr;
+nav2_behavior_tree::DistanceController * DistanceControllerTestFixture::node_ = nullptr;
+nav2_behavior_tree::DummyNode * DistanceControllerTestFixture::dummy_node_ = nullptr;
+
+TEST_F(DistanceControllerTestFixture, test_initial_status_is_idle)
+{
+  EXPECT_EQ(node_->status(), BT::NodeStatus::IDLE);
+}
+
+TEST_F(DistanceControllerTestFixture, test_failure_when_child_is_idle)
+{
+  dummy_node_->setStatus(BT::NodeStatus::IDLE);
+  EXPECT_EQ(node_->executeTick(), BT::NodeStatus::FAILURE);
+}
+
+TEST_F(DistanceControllerTestFixture, test_failure_when_child_fails)
+{
+  dummy_node_->setStatus(BT::NodeStatus::FAILURE);
+  EXPECT_EQ(node_->executeTick(), BT::NodeStatus::FAILURE);
+}
+
+TEST_F(DistanceControllerTestFixture, test_running_when_child_is_running)
+{
+  dummy_node_->setStatus(BT::NodeStatus::RUNNING);
+  EXPECT_EQ(node_->executeTick(), BT::NodeStatus::RUNNING);
+}
+
+TEST_F(DistanceControllerTestFixture, test_behavior)
+{
+  EXPECT_EQ(node_->status(), BT::NodeStatus::IDLE);
+
+  dummy_node_->setStatus(BT::NodeStatus::SUCCESS);
+  EXPECT_EQ(node_->executeTick(), BT::NodeStatus::SUCCESS);
+  EXPECT_EQ(dummy_node_->status(), BT::NodeStatus::IDLE);
+
+  geometry_msgs::msg::PoseStamped pose;
+  pose.pose.position.x = 0.5;
+  pose.pose.position.y = 0;
+  pose.pose.position.z = 0;
+  pose.pose.orientation.x = 0;
+  pose.pose.orientation.y = 0;
+  pose.pose.orientation.z = 0;
+  pose.pose.orientation.w = 1;
+
+  double traveled = 0;
+
+  transform_handler_->updateRobotPose(pose.pose);
+
+  // Wait for transforms to actually update
+  while (traveled < 0.1) {
+    if (nav2_util::getCurrentPose(pose, *transform_handler_->getBuffer())) {
+      traveled = std::sqrt(
+        pose.pose.position.x * pose.pose.position.x + pose.pose.position.y * pose.pose.position.y);
+    }
+    EXPECT_EQ(node_->executeTick(), BT::NodeStatus::RUNNING);
+  }
+
+  for (int i = 1; i < 10; i++) {
```

Comment:
So I think my understanding of this is that you're testing in each `i` that it moves enough to cause an update. I think you should double this to `20` and then half the `1.1` so that you're checking intermediate distances that they return FAILURE when it hasn't moved enough. That will make sure that it works for not enough and enough main cases.